sled-agent/src/sim/instance.rs

```diff
@@ -87,12 +192,28 @@ impl Simulatable for SimInstance {
         self.state.current()
     }
 
-    fn desired(&self) -> &Option<Self::RequestedState> {
-        self.state.desired()
+    fn desired(&self) -> Option<Self::RequestedState> {
+        self.propolis_queue.back().map(|terminal| match terminal {
+            PropolisInstanceState::Creating
+            | PropolisInstanceState::Starting
+            | PropolisInstanceState::Migrating
+            | PropolisInstanceState::Repairing
+            | PropolisInstanceState::Failed => panic!(
+                "terminal state {:?} doesn't map to a requested state",
+                terminal
+            ),
```

Comment:
The safety on this is just that "whenever we `push_back` to the `propolis_queue`, we don't *end* with these states, right?

Reply:
That's right--a state change request might queue one of these states as a transitional state, but it should never be the very last state on the queue. This is supposed to be an internal invariant in this module, hence the panic. I'll add a comment clarifying this.